### PR TITLE
Modify PyInstaller's build to avoid encoding project path names.

### LINKED ISSUE #26: Some error messages refer to paths from buildtime
In issue #25, I managed to get errors like:
```
Traceback (most recent call last):
  File "C:\projects\pyexe\pyexe.py", line 1176, in <module>
  File "C:\Python36\lib\os.py", line 1017, in fdopen
ValueError: Must have exactly one of read or write mode
[13316] Failed to execute script pyexe
```
These refer to "C:\projects\" and "C:\Python36".  This could be especially confusing if user actually has those directories...

### PR DESCRIPTION
**Pull request by manthey**
Resolves #26.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -53,8 +53,9 @@ build_script:
   - del /S /Q %PYTHON%\*.pyc >NUL
   - del /S /Q %PYTHON%\*.pyo >NUL
   - python modules_pyexe.py pyexe.py
-  # Install PyInstaller
-  - python -m pip install pyinstaller
+  # Install PyInstaller.  We patch various aspects of PyInstaller, so use a 
+  # fixed version so we know
```
Comment (cbm755) [inline appveyor.yml]: comment incomplete

File: appveyor.yml [hunk 2]
```diff
@@ -64,6 +65,8 @@ build_script:
   - "copy hooks\\pyi_rth*.* %PYTHON%\\Lib\\site-packages\\PyInstaller\\loader\\rthooks\\."
   # Replace the stage 3 importer
   - "copy hooks\\pyimod03_importers.py %PYTHON%\\Lib\\site-packages\\PyInstaller\\loader\\."
+  # Replace the part of the build script to avoid storing project path names
+  - "copy hooks\\utils.py %PYTHON%\\Lib\\site-packages\\PyInstaller\\building\\."
```
Comment (cbm755) [inline appveyor.yml]: Am I right that you're dropping in a modified version of an existing utils.py?  Why not patch it instead?

File: hooks/utils.py [hunk 3]
```diff
@@ -0,0 +1,642 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2017, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License with exception
+# for distributing bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
```
Comment (cbm755) [inline hooks/utils.py]: Will this mean changing the license of this project?  (I have no problem with the GPL, just pointing it out).

Patching this file would avoid this.

Comment (manthey) [reply hooks/utils.py]: Thanks for pointing this out (I'll plead that I was doing this at an odd hour and failed to think things through).  I'll change this to patching (and also fix the comment typo).

**Pull request merged**